src/Fractions/Formatter/DecimalNotationFormatter.cs

```diff
@@ -235,25 +235,23 @@ public static string Format(string? format, Fraction fraction, IFormatProvider?
         };
     }
 
-
-    private static int GetPrecisionDigitsOrDefault(string format, int defaultValue) {
-        return format.Length > 1 ? GetPrecisionDigits(format) : defaultValue;
-    }
-
-    private static int GetPrecisionDigits(string format) {
- #if NET
-        if (!int.TryParse(format.AsSpan(1), out var nbDecimals) || nbDecimals < 0) {
-            throw new FormatException($"The {format} format string is not supported.");
-        }
+    private static bool TryGetPrecisionDigits(string format, int defaultPrecision, out int maxNbDecimals) {
+        if (format.Length != 1) {
+#if NET
+            if (!int.TryParse(format.AsSpan(1), out maxNbDecimals)) {
+                return false;
+            }
 #else
-        if (!int.TryParse(format.Substring(1), out var nbDecimals) || nbDecimals < 0) {
-            throw new FormatException($"The {format} format string is not supported.");
-        }
+            if (!int.TryParse(format.Substring(1), out maxNbDecimals)) {
+                return false;
+            }
 #endif
+        }
 
-        return nbDecimals;
+        maxNbDecimals = defaultPrecision;
+        return true;
     }
-
+    
```

Comment:
I've added some changes.
![image](https://github.com/user-attachments/assets/b5fe55fd-17b2-426b-adaa-7cdc6de36520)


```suggestion
        private static bool TryGetPrecisionDigits(string format, int defaultPrecision, out int maxNbDecimals) {
        if (format.Length == 1) {
            // The number of digits is not specified, use default precision.
            maxNbDecimals = defaultPrecision;
            return true;
        }

#if NET
        if (int.TryParse(format.AsSpan(1), out maxNbDecimals)) {
            return true;
        }
#else
        if (int.TryParse(format.Substring(1), out maxNbDecimals)) {
            return true;
        }
#endif

        // Seems to be some kind of custom format we do not understand, fallback to default precision.
        maxNbDecimals = defaultPrecision;
        return false;
    }
```